sql/plan/resolved_table.go

```diff
@@ -196,6 +210,32 @@ func (t *ResolvedTable) WithTable(table sql.Table) (*ResolvedTable, error) {
 		return nil, fmt.Errorf("attempted to update TableNode `%s` with table `%s`", t.Name(), table.Name())
 	}
 	nt := *t
+
 	nt.Table = table
 	return &nt, nil
 }
+
+// WithWrappedTable returns this Node with the given table, re-wrapping it with any MutableTableWrapper that was
+// wrapping it prior to this call.
+func (t *ResolvedTable) WithWrappedTable(table sql.Table) (*ResolvedTable, error) {
+	if t.Name() != table.Name() {
+		return nil, fmt.Errorf("attempted to update TableNode `%s` with table `%s`", t.Name(), table.Name())
+	}
+	nt := *t
+
+	if mtw, ok := nt.Table.(sql.MutableTableWrapper); ok {
+		nt.Table = mtw.WithUnderlying(table)
+	} else {
+		nt.Table = table
+	}
+
+	return &nt, nil
+}
+
+func (t *ResolvedTable) ReWrapTable(table sql.IndexedTable) (sql.IndexedTable, error) {
```

Comment:
the difference between ReWrap and WithWrapped isn't immediately obvious, just a shorthand to avoid the sql.IndexedTable casting on the other side of the call?